src/include/distributed/distributed_planner.h

```diff
@@ -85,10 +85,31 @@ typedef struct JoinRestriction
 	RelOptInfo *outerrel;
 } JoinRestriction;
 
+typedef struct FastPathRestrictionContext
+{
+	bool fastPathRouterQuery;
+
+	/*
```

Comment:
This comment only mentions prepared statements, however we also do not set it if the dist key column type doesn't match the const's type. So, either expand the comment or do not mention about the specific cases where distributionKeyValue == NULL